### PR TITLE
Feature ETP-83: Data Handling Function for Backend Retrieval

### PR DESCRIPTION
**Pull request by LuuchoRocha**
## Feature ETP-83

Updated Datasource module and useDatasource hook with the new requirements.
It accepts the following parameters:

```ts
interface DatasourceOptions {
  columns?: string[];
  sortBy?: string;
  criteria?: Criteria[];
  operationType?: 'fetch' | 'add' | 'update' | 'remove';
  isSorting?: boolean;
  isImplicitFilterApplied?: boolean;
  operator?: 'and' | 'or';
}

interface Criteria {
  fieldName: string;
  operator: string;
  value: string;
}
```

Example:

[Grabación de pantalla desde 2024-08-15 14-08-25.webm](https://github.com/user-attachments/assets/6a750150-764a-42f8-87b3-73da250fe802)

### CODE REVIEW

File: packages/ComponentLibrary/src/theme.ts [hunk 1]
```diff
@@ -89,12 +89,12 @@ export const theme = createTheme({
       dark: SECONDARY_300,
       contrastText: PRIMARY_500,
     },
-    tertiary: {
-      light: TERTIARY_50,
-      main: TERTIARY_50,
-      dark: TERTIARY_100,
-      contrastText: PRIMARY_500,
-    },
+    // tertiary: {
```
Comment (sjcuello) [inline packages/ComponentLibrary/src/theme.ts:92]: Use the colors defined in figma.

File: packages/ComponentLibrary/src/components/index.ts [hunk 2]
```diff
@@ -26,37 +23,34 @@ import SecondaryTabsMUI from './SecondaryTabs';
 import NotificationItemStates from './NotificationItemAllStates';
 import DrawerMUI from './Drawer';
 import IButton from './IconButton';
-import { BoxProps } from '@mui/material';
 
 export * from '../theme';
 
-// ApplyHOC is a higher order component that applies the theme to the component
+//  is a higher order component that applies the theme to the component
```
Comment (sjcuello) [inline packages/ComponentLibrary/src/components/index.ts:29]: Why is the reason to not apply the HOC with the styles ?

Comment (LuuchoRocha) [reply packages/ComponentLibrary/src/components/index.ts:29]: The HOC wraps each component in its own ThemeProvider, but the ThemeProvider is already wrapping the entire App component, so, it is not need anymore

File: packages/ComponentLibrary/src/theme.ts (hunk 1 above)
Comment (LuuchoRocha) [reply packages/ComponentLibrary/src/theme.ts]: I was trying to fix the build and the existing theme interface did not have "tertiary" prop and TS was complaining about it. I reverted back this change and updated the Theme and Palette type definitions

Comment (sjcuello) [reply packages/ComponentLibrary/src/theme.ts]: This configuration was already working, that's why I asking.
Please check with S.Alaniz if this change touch other component.
@Sankku

Comment (LuuchoRocha) [reply packages/ComponentLibrary/src/theme.ts]: Yep I saw you've added the custom interface but somehow it got broken.
About this fix, we just checked together and he approved it

Comment (sjcuello) [reply packages/ComponentLibrary/src/theme.ts]: Okay, well done !

**Review by sjcuello — APPROVED** [review]

**Review by tomyrodeghiero — APPROVED** [review]

**Pull request merged**